pkg/rehearse/jobs.go

```diff
@@ -163,11 +174,18 @@ func inlineCiOpConfig(job *prowconfig.Presubmit, targetRepo string, ciopConfigs
 	return &rehearsal, nil
 }
 
-func configureRehearsalJobs(toBeRehearsed map[string][]prowconfig.Presubmit, prRepo string, prNumber int, loggers Loggers) []*prowconfig.Presubmit {
+// ConfigureRehearsalJobs filters the jobs that should be rehearsed, then return a list of them re-configured with the
+// ci-operator's configuration inlined.
+func ConfigureRehearsalJobs(toBeRehearsed map[string][]prowconfig.Presubmit, prRepo string, prNumber int, loggers Loggers, allowVolumes bool) []*prowconfig.Presubmit {
+	rehearsalsFiltered := filterJobs(toBeRehearsed, allowVolumes, loggers.Job)
+	if len(rehearsalsFiltered) == 0 {
+		return nil
```

Comment:
Why? Doesn't `filterJobs` return `nil`?

Reply:
While filtering, some jobs can be excluded. Here, we make sure that if we exclude all the jobs, not to continue.

Reply:
Not a fan of this or the other change you have made to explicitly exit early on empty slices. `for` loop over a slice will do nothing if there is nothing in the slice. No need to have this be explicit.

Reply:
Yes, good point. I will change it.

Reply:
addressed